setup/CMakeLists.txt

```diff
@@ -0,0 +1,219 @@
+
+ExternalProject_Add(grpc
+  GIT_REPOSITORY  https://github.com/google/grpc.git
+  GIT_TAG         ${GRPC_GIT_TAG}
+  GIT_SHALLOW     ON    # depth=1
+  GIT_PROGRESS    ON
+
+  SOURCE_DIR
+    ${CMAKE_SOURCE_DIR}/grpc
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+    -DCMAKE_POSITION_INDEPENDENT_CODE=on
+    -DBUILD_SHARED_LIBS=on
+    -DgRPC_ABSL_PROVIDER=package
+    -DgRPC_PROTOBUF_PROVIDER=package
+    -DgRPC_BUILD_GRPC_CSHARP_PLUGIN=off
+    -DgRPC_BUILD_GRPC_NODE_PLUGIN=off
+    -DgRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN=off
+    -DgRPC_BUILD_GRPC_PHP_PLUGIN=off
+    -DgRPC_BUILD_GRPC_RUBY_PLUGIN=off
+    -DgRPC_BUILD_TESTS=off
+    -DgRPC_INSTALL=on
+  INSTALL_COMMAND
+    ${SUDO_CMD} ${CMAKE_MAKE_PROGRAM} install
+  DEPENDS
+    abseil-cpp
+    protobuf
+)
+
+########
+# CCTZ #
+########
+
+set(CCTTZ_GIT_TAG 02918d62329ef440935862719829d061a5f4beba)
+
+ExternalProject_Add(cctz
+  GIT_REPOSITORY  https://github.com/google/cctz.git
+  GIT_TAG         ${CCTZ_GIT_TAG}
+  GIT_PROGRESS    ON
+
+  SOURCE_DIR
+    ${CMAKE_SOURCE_DIR}/cctz
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+    -DCMAKE_POSITION_INDEPENDENT_CODE=on
+    -DBUILD_TESTING=off
+  INSTALL_COMMAND
+    ${SUDO_CMD} ${CMAKE_MAKE_PROGRAM} install
+)
+
+##########
+# GFLAGS #
+##########
+
+set(GFLAGS_GIT_TAG 827c769e5fc98e0f2a34c47cef953cc6328abced)
+
+ExternalProject_Add(gflags
+  GIT_REPOSITORY  https://github.com/gflags/gflags.git
+  GIT_TAG         ${GFLAGS_GIT_TAG}
+  GIT_PROGRESS    ON
+
+  SOURCE_DIR
+    ${CMAKE_SOURCE_DIR}/gflags
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+    -DBUILD_SHARED_LIBS=on
+  INSTALL_COMMAND
+    ${SUDO_CMD} ${CMAKE_MAKE_PROGRAM} install
+)
+
+########
+# GLOG #
+########
+
+set(GLOG_GIT_TAG 503e3dec8d1fe071376befc62119a837c26612a3)
+
+ExternalProject_Add(glog
+  GIT_REPOSITORY  https://github.com/google/glog.git
+  GIT_TAG         ${GLOG_GIT_TAG}
+  GIT_PROGRESS    ON
+
+  SOURCE_DIR
+    ${CMAKE_SOURCE_DIR}/glog
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+    -Dgflags_DIR:PATH=${CMAKE_INSTALL_PREFIX}/lib/cmake/gflags
+    -DWITH_GTEST=OFF
+  INSTALL_COMMAND
+    ${SUDO_CMD} ${CMAKE_MAKE_PROGRAM} install
+  DEPENDS
+    gflags
+)
+
+#############
+# GMOCK_GBL #
+#############
+
+ExternalProject_Add(gmock-gbl
+  GIT_REPOSITORY  https://github.com/apriorit/gmock-global-sample.git
+  GIT_PROGRESS    ON
+
+  SOURCE_DIR
+    ${CMAKE_SOURCE_DIR}/gmock-gbl
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+)
+
+#########
+# GTEST #
+#########
+
+set(GTEST_GIT_TAG release-1.11.0)
+
+ExternalProject_Add(gtest
+  GIT_REPOSITORY  https://github.com/google/googletest.git
+  GIT_TAG         ${GTEST_GIT_TAG}
+  GIT_PROGRESS    ON
+
+  SOURCE_DIR
+    ${CMAKE_SOURCE_DIR}/gtest
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+    -DBUILD_SHARED_LIBS=on
+  INSTALL_COMMAND
+    ${SUDO_CMD} ${CMAKE_MAKE_PROGRAM} install
+)
```

Comment:
Maybe we don't need gtest and gmock_gbl, we can add these when add support for unit tests?

Reply:
I included googletest and gmock-global because they were part of the existing package.

Incidentally, the mock-global package installation is defective. It doesn't add anything to the install tree!

My inclination is to:
1.  Make googletest and gmock-global conditional, under control of CMake options.
2.  Install `include/gmock-global/gmock-global.h` if gmock-gbl is enabled.

Long-term, we need to differentiate between development installations (which would normally _include_ header files and support for unit tests) and target installations (which would normally _omit_ header files and support for unit testing).

Reply:
I like the cmake option idea and ability to differentiate development and target installations. We can add this when adding support for different build types